File: Deliverytime.py
import pandas as pd
import numpy as np
delivery=pd.read_csv("C://Users//POOVAYUVA//Desktop//EXCELR ASSIGN//SIMPLE LINEAR//delivery_time.csv")
delivery.plot.box()


delivery.plot.hist()

deliverycorr=delivery.corr()

deliverycorr

import statsmodels.formula.api as sm

model=sm.ols("Delivery~Sorting",data=delivery).fit()

model.summary()

Predicted=model.predict()

delivery["log_Sorting"] = np.log(delivery.Sorting)

delivery.head()

model2=sm.ols("Delivery~log_Sorting",data=delivery).fit()

model2.summary()

predict2=model2.predict()

delivery["log_Delivery"] = np.log(delivery.Delivery)

delivery.head()

model3=sm.ols("log_Delivery~Sorting",data=delivery).fit()

model3.summary()

predict3=model3.predict()

model4=sm.ols("log_Delivery~log_Sorting",data=delivery).fit()

model4.summary()

predict4=model4.predict()

delivery["x2"] = delivery.Sorting ** 2

delivery.head()

model5=sm.ols("log_Delivery~Sorting + x2",data=delivery).fit()

model5.summary()

predict5=model5.predict()

def RMSE(Pred, Act):

    sq_error=(Pred-Act)**2

    mean_sqerror=np.mean(sq_error)
    root_mean_sqerror=np.sqrt(mean_sqerror)
    return(root_mean_sqerror)
print(RMSE(Predicted,delivery.Delivery))

print(RMSE(predict2,delivery.Delivery))

print(RMSE(np.exp(predict3),delivery.Delivery))

print(RMSE(np.exp(predict4),delivery.Delivery))

print(RMSE(np.exp(predict5),delivery.Delivery))

np.mean(Predicted-delivery.Delivery)

np.mean(predict2-delivery.Delivery)

File: delivery time.R
library(readr)
delivery_time <- read_csv("C:/Users/POOVAYUVA/Desktop/EXCELR ASSIGN/SIMPLE LINEAR/delivery_time.csv")
View(delivery_time)

attach(delivery_time)

plot(delivery_time$`Delivery Time`,delivery_time$`Sorting Time`)

cor.test(delivery_time$`Delivery Time`,delivery_time$`Sorting Time`) #0.82

dt <- lm(delivery_time$`Delivery Time`~delivery_time$`Sorting Time`,data =delivery_time)
summary(dt) #0.6655

error=(delivery_time$`Delivery Time`- dt$fitted.values)
rmsen=sqrt(mean(error^2))


#logrithmic transformation

attach(delivery_time)
plot(log(`Sorting Time`),`Delivery Time`)
cor.test(log(`Sorting Time`),`Delivery Time`) #0.83 
dtl <- lm(`Delivery Time`~log(`Sorting Time`),data=delivery_time)
summary(dtl)  #0.6794
error=(`Delivery Time`- dtl$fitted.values)
rmsel=sqrt(mean(error^2))


#exponential
attach(delivery_time)

cor.test(`Sorting Time`,log(`Delivery Time`)) #0.8431
dte <- lm(log(`Delivery Time`)~`Sorting Time`,data=delivery_time)
summary(dte) #0.6957
exp(dte$fitted.values)
exerroe <-`Delivery Time`-exp(dte$fitted.values)
rmsee <- sqrt(mean(exerroe^2))


#quadratic equation
plot(`Sorting Time`,`Delivery Time`)
plot(`Sorting Time`^2,`Delivery Time`)
plot (`Sorting Time`^2,log(`Delivery Time`)
cor.test(`Sorting Time`^2,log(`Delivery Time`)) #0.7882452
qddt <- lm(log(`Delivery Time`)~`Sorting Time`+I(`Sorting Time`^2),data=delivery_time)
summary (qddt) #0.7387

exp(qddt$fitted.values)
errorq <- `Delivery Time`-exp(qddt$fitted.values)
rmseq <- sqrt(mean(errorq^2)) #2.799

#since quadratic transformation gives the better Adjusted R-squared value=0.7387
#hence qddt will be the better model.

